tests/bfd/test_bfd.py

```diff
@@ -481,3 +490,35 @@ def test_bfd_multihop(request, rand_selected_dut, ptfhost, tbinfo,
         duthost.shell(cmd_buffer, module_ignore_errors=True)
         ptfhost.command('supervisorctl stop bfd_responder')
         ptfhost.file(path=BFD_RESPONDER_SCRIPT_DEST_PATH, state="absent")
+
+
+@pytest.mark.parametrize('ipv6', [False, True], ids=['ipv4', 'ipv6'])
+def test_bfd_echo_mode(request, rand_selected_dut, ptfhost, tbinfo, ipv6):
+    duthost = rand_selected_dut
+    bfd_session_cnt = int(request.config.getoption('--num_sessions'))
+
+    # Get neighbors for BFD sessions
+    neighbor_addrs = get_neighbors(duthost, tbinfo, ipv6, count=bfd_session_cnt)[2]
+
+    try:
+        # Use bfd_echo_mode function for direct configuration
+        bfd_echo_mode(duthost, neighbor_addrs)  # Pass None for optional logger
+
+        # Verify BFD sessions with echo mode enabled
+        time.sleep(30)  # Wait for BFD sessions to be established
+        result = duthost.shell("vtysh -c 'show bfd peer'")['stdout'].split('\n')
+        error = False
+        for line in result:
+            if ('Echo transmission interval:' in line) or (
+                    'Echo receive interval:' in line):
+                if 'disabled' in line:
+                    error = True
+        assert error is False
+    finally:
+        # Cleanup: Remove BFD sessions and echo mode configurations
+        remove_bfd_sessions(duthost, neighbor_addrs)
```

Comment:
if no sessions are created why the cleanup.

Reply:
BFD sessions are being created first as mentioned in the previous comment. So, cleanup is required for session tear-down.